runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaTimerInternalsFactory.java

```diff
@@ -702,8 +673,14 @@ public void encode(TimerKey<K> value, OutputStream outStream)
       if (keyCoder != null) {
         key = keyCoder.decode(inStream);
       }
+      timerKeyBuilder.setTimerId(timerId).setStateNamespace(namespace).setKey(key);
+
+      // check if the stream has more available bytes. This is to ensure backward compatibility with
+      // old rocksdb state which does not encode timer family data
+      final String timerFamilyId = inStream.available() > 0 ? STRING_CODER.decode(inStream) : "";
+      timerKeyBuilder.setTimerFamilyId(timerFamilyId);
 
-      return new TimerKey<>(key, namespace, timerId);
+      return timerKeyBuilder.build();
```

Comment:
nit: we do not need to declare the builder at the begining

we can do 

```
return TimerKey.builder().setX().setY().setZ().build();
```

Reply:
RIght, it seems nicer to have the builder in the end instead of cutting it in half.

Reply:
Changed it.